src/unix/linux_like/android/mod.rs

```diff
@@ -1020,6 +1034,31 @@ cfg_if! {
                     .finish()
             }
         }
+
+        impl Eq for ethhdr {}
+
+        impl PartialEq for ethhdr {
+            fn eq(&self, other: &ethhdr) -> bool {
+                self.h_dest
+                    .iter()
+                    .zip(other.h_dest.iter())
+                    .all(|(a, b)| a == b)
```

Comment:
Do you not want them to compare equal if they have the same CStr value, but different null trailing bytes?

Reply:
note that those fields represent mac addresses not as strings (note they re unsigned char).

Reply:
Ah, my bad, you're correct.